upstream/upstream_test.go

```diff
@@ -0,0 +1,333 @@
+
+		var events []api.Event
+		err = testDB.Where("name = ? AND created_at >= ?", pkgEvents.EventPushQueueCreate, start).Find(&events).Error
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(len(events)).To(Equal(3))
+
+		groupedEvents := pkgEvents.GroupChangelogsByTables(events)
+		Expect(groupedEvents["components"]).To(Equal([][]string{{modifiedNewDummy.ID.String()}, {modifiedNewDummy.ID.String()}, {modifiedNewDummy.ID.String()}}))
+	})
+
+	ginkgo.It("should transfer all events to upstream server", func() {
+		// Overwrite the global db that'll be used by the upstream server.
+		db.Gorm = testUpstreamDB
+		db.Pool = testUpstreamDBPGPool
+
+		eventHandlerConfig := events.Config{
+			UpstreamConf: api.UpstreamConfig{
+				ClusterName: "test-cluster",
+				URL:         fmt.Sprintf("http://localhost:%d/upstream_push", testUpstreamServerPort),
+				Username:    "admin@local",
+				Password:    "admin",
+				Labels:      []string{"test"},
+			},
+		}
+
+		events.ConsumeEventsUntilEmpty(context.Background(), testDB, eventHandlerConfig)
+	})
+
+	ginkgo.It("should have transferred all the components", func() {
+		var fieldsToIgnore []string
+		fieldsToIgnore = append(fieldsToIgnore, "SystemTemplateID")                                              // Upstream creates its own dummy template
+		fieldsToIgnore = append(fieldsToIgnore, "Checks", "Components", "Order", "SelectorID", "RelationshipID") // These are auxiliary fields & do not represent the table columns.
+		ignoreFieldsOpt := cmpopts.IgnoreFields(models.Component{}, fieldsToIgnore...)
+
+		// unexported fields must be explicitly ignored.
+		ignoreUnexportedOpt := cmpopts.IgnoreUnexported(models.Component{}, models.Summary{})
+
+		compareEntities(testUpstreamDB, testDB, &[]models.Component{}, ignoreFieldsOpt, ignoreUnexportedOpt)
+	})
+
+	ginkgo.It("should have transferred all the checks", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.Check{})
+	})
+
+	ginkgo.It("should have transferred all the check statuses", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.CheckStatus{})
+	})
+
+	ginkgo.It("should have transferred all the canaries", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.Canary{})
+	})
+
+	ginkgo.It("should have transferred all the configs", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ConfigItem{})
+	})
+
+	ginkgo.It("should have transferred all the config analyses", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ConfigAnalysis{})
+	})
+
+	ginkgo.It("should have transferred all the config relationships", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ComponentRelationship{})
+	})
+
+	ginkgo.It("should have transferred all the config component relationships", func() {
+		compareEntities(testUpstreamDB, testUpstreamDB, &[]models.ConfigComponentRelationship{})
+	})
+})
+
+func populateMonitoredTables(gormDB *gorm.DB) error {
```

Comment:
Is there any drawback to calling the populate all tables function from duty ?

Reply:
I don't think there's any. I initially started with duty's dummy for a quick test but then stuck with it as that worked fine.

Reply:
Use this function, takes db instance and loads all the dummy data
```
fixtures/dummy/all.go
9:func PopulateDBWithDummyModels(gormDB *gorm.DB) error {
```